### ISSUE TITLE
[BUG] The type initializer for `Azure.Search.Documents.Constants` threw an exception.

### ISSUE THREAD

**Issue by 12c-dk** [Azure/azure-sdk-for-net#22718]
**Describe the bug**
I had a problem getting Exceptions when using the Azure.Search.Documents 11.2.1 NuGet package. I'm working on an Azure Function integration project using .net core 3.1 and Azure Functions v. 3. 

When developing and debugging locally, everything works fine, but when publishing to the Azure function and calling the endpoints I'm getting Exceptions with the message: "The type initializer for 'Azure.Search.Documents.Constants' threw an exception."

This happens when executing a search call like this:

SearchResults<ProjectSearchResults> response = await _searchClient.SearchAsync<ProjectSearchResults>(queryText, options);

**Expected behavior**
I expected that the search would succeed in the Azure Function App, in the same way it does on my local development environment.

**Actual behavior (include Exception or Stack Trace)**
Exception with the message text is thrown: "The type initializer for 'Azure.Search.Documents.Constants' threw an exception."

When investigating my Application insights logs further, i found the following from the stack trace:

System.MissingMethodException: Method not found: 'System.Text.Json.JsonEncodedText System.Text.Json.JsonEncodedText.Encode(System.String, System.Text.Encodings.Web.JavaScriptEncoder)'.
   at Azure.Search.Documents.Constants..cctor()

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Setup an Azure function v3 project using .net core 3.1
2. Add NuGet package: Azure.Search.Documents 11.2.1
3. Setup search client
3. Setup a Http triggered Azure function
4. Make call to search using: SearchResults<ProjectSearchResults> response = await _searchClient.SearchAsync<ProjectSearchResults>(queryText, options);
6. Publish to Azure Function App
7. Invoke Http triggered Azure function

**Environment:**
 - Name and version of the Library package used: Azure.Search.Documents 11.2.1
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): Azure Function App v3, .net core 3.1. 
 - IDE and version : Microsoft Visual Studio Professional 2019 - Version 16.8.6

**Suggested fix**
I can see that the Azure.Search.Documents NuGet package has a dependency to System.Text.Json >= 4.6.0

Azure.Search.Documents is using a method that is implemented in a later version of System.Text.Json. I fixed the issue in my project by manually installing System.Text.Json v5.0.2. 

I recommend that you update System.Text.Json depencency in the Azure.Search.Documents to version 5.0.2 or a version that implements the missing method.

**Comment by lumichael24** [comment]
I'm facing this same problem with Azure.Search.Documents 11.3.0 NuGet package

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by Mohit-Chakraborty** [comment]
Some comments from what I see -
* All Azure track2 SDK packages including `Azure.Search.Documents` reference `System.Text.Json` v4.6.0. An accidental dependency on a higher version is unlikely.
* `Azure.Search.Documents.Constants` did not change for a long time leading up to 11.2.1 release.

This leads me to believe the issue is in the intersection of the SDK and Azure Functions.

**Comment by 12c-dk** [comment]
Thank you for looking into it. You might be right. But even though there might be a problem in the Function app tools, i think that this issue could be resolved by updating the NuGet dependencies. 

I have tested this a bit further on my project and it turns out that F5 debugging my function app also throws the error locally, the same as when running in Azure Function App. I have upgraded and downgraded my System.Text.Json Nuget package and i am consistently getting exceptions on the following versions of System.Text.Json:
4.7.0
4.6.0
And lower

I also verified that it consistently works on the following versions of System.Text.Json
5.0.2
5.0.0
4.7.2
4.7.1

Another thing i found was that the System.Text.Json.dll file is a part of the build output when running System.Text.Json versions 4.7.1 and highger, but using 4.7.0 and previous versions, the System.Text.Json.dll is not a part of the build output. 

This is the exception output i'm getting when running the project locally using System.Text.Json version 4.7.0 and below.

[2021-07-20T19:33:14.302Z] SearchProjects: Failed to get projects from search.: The type initializer for 'Azure.Search.Documents.Constants' threw an exception.    at Azure.Search.Documents.Models.SearchResults`1.DeserializeAsync(Stream json, ObjectSerializer serializer, Boolean async, CancellationToken cancellationToken)
[2021-07-20T19:33:14.304Z]    at Azure.Search.Documents.SearchClient.SearchInternal[T](SearchOptions options, String operationName, Boolean async, CancellationToken cancellationToken)
[2021-07-20T19:33:14.306Z]    at Azure.Search.Documents.SearchClient.SearchInternal[T](String searchText, SearchOptions options, Boolean async, CancellationToken cancellationToken)
[2021-07-20T19:33:14.307Z]    at Azure.Search.Documents.SearchClient.SearchAsync[T](String searchText, SearchOptions options, CancellationToken cancellationToken)
[2021-07-20T19:33:14.308Z]    at Portal.Project.Common.DataSvc.Repositories.SearchRepository.Search(Search searchRequest) in C:\dev\Portal.Project.Common.DataSvc\Src\Portal.Project.Common.DataSvc\Repositories\SearchRepository.cs:line 72
[2021-07-20T19:33:14.308Z]    at Portal.Project.Common.DataSvc.Functions.SearchProjects.Run(HttpRequest req, ILogger log) in C:\dev\Portal.Project.Common.DataSvc\Src\Portal.Project.Common.DataSvc\Functions\SearchProjects.cs:line 49

**Comment by Mohit-Chakraborty** [comment]
Thanks for providing additional details.
> Another thing i found was that the System.Text.Json.dll file is a part of the build output when running System.Text.Json versions 4.7.1 and highger, but using 4.7.0 and previous versions, the System.Text.Json.dll is not a part of the build output.

If you manually copy the missing DLL to the output directory, does that impact the behavior?

> I have tested this a bit further on my project and it turns out that F5 debugging my function app also throws the error locally, the same as when running in Azure Function App.

Good to know that the error can be reproduced locally. Have you tried any other type of SDK package with your Azure Functions app? I wonder if we see the same issue with other libraries.

**Comment by 12c-dk** [comment]
Thank you for the tip. 

I tried to copy in the System.Text.Json.dll as you suggested. First i installed the System.Text.Json NuGet in version 4.7.0, build the project and copied the System.Text.Json.dll file to the build output. It seems that the dll isn't loaded and the outcome was the same as before, getting the same Exception. 

Then i tried installing the System.Text.Json NuGet in version 4.7.1, build the project and replacing the dll with the previous 4.7.0 version, but when i start debugging the file is replaced. If i try to update it while the project is running it is detected by Visual studio and the project stops. 

I noticed a difference in the function.deps.json files, in the targets.".NETCoreApp,Version=v3.1", when running with the 4.7.0 and the 4.7.1 versions of the System.Text.Json package. With the 4.7.0 package installed, the System.Text.Json element looks like this:

"System.Text.Json/4.7.0": {},

With the 4.7.1 package it looks like this:

"System.Text.Json/4.7.1": {
        "runtime": {
          "lib/netcoreapp3.0/System.Text.Json.dll": {
            "assemblyVersion": "4.0.1.1",
            "fileVersion": "4.700.20.6702"
          }
        }
      },

I'm not sure what you mean by another type of SDK package with my Azure functions app. Do you mean the Microsoft.NET.Sdk.Functions package i'm using in the project? I didn't think of this, thank you. I can see that i'm running on an old 3.0.1 version of the Microsoft.NET.Sdk.Functions. Upgrading this often requires upgrade of several dependent packages. I will have a go at this tomorrow.

**Comment by 12c-dk** [comment]
Thank you for your advice on upgrading the Functions SDK package. 

I just upgraded the Microsoft.NET.Sdk.Functions package of my project to version 3.0.13 and now it works. I tried manually installing System.Text.Json 4.7.0 and 4.7.1 and search works in both cases. I tried uninstalling the System.Text.Json package and it still works. I published it to Azure and executed the Azure function and it also works here. 

So you are right, it looks like there is a problem with the Microsoft.NET.Sdk.Functions 3.0.1. Thank you for your advice!
